src/Infrastructure/Util/src/ESMF_LogErr.F90

```diff
@@ -1730,11 +1730,80 @@ subroutine ESMF_LogInitialize(filename, logappendflag, logkindflag, rc)
 !      \end{description}
 !
 !EOPI
+
+    integer                        :: localrc
+    integer                        :: msgAbortCnt
+    type(ESMF_LogMsg_Flag)         :: msgAbortLst(6)
+    character(ESMF_MAXSTR)         :: envRtAbort
+    integer                        :: idxRtAbort
+    type(ESMF_LogPrivate), pointer :: alog
+
     ! Initialize return code; assume routine not implemented
     if (present(rc)) rc = ESMF_RC_NOT_IMPL
 
     call ESMF_LogOpen(ESMF_LogDefault, filename,  &
-        appendflag=logappendflag, logkindflag=logkindflag, rc=rc)
+        appendflag=logappendflag, logkindflag=logkindflag, rc=localrc)
+    if (localrc /= ESMF_SUCCESS) then
+      write (ESMF_UtilIOStderr,*) ESMF_METHOD,  &
+          ": ESMF_LogOpen - ESMF_LogDefault"
+      call ESMF_UtilIOUnitFlush(ESMF_UtilIOStderr, rc=rc)
+      if (present(rc)) rc=localrc
+      return
+    end if
```

Comment:
Should use standard error handling via `ESMF_LogFoundError()`. I think it is historical that this file isn't doing that consistently. Let's at least do it for new code that being added to this file.

```suggestion
    if (ESMF_LogFoundError(localrc, ESMF_ERR_PASSTHRU, &
        ESMF_CONTEXT, rcToReturn=rc)) return
```